### ISSUE TITLE
Errors unless `FORUM_MONGODB_DATABASE` and `FORUM_MONGODB_CLIENT_PARAMETERS` are defined

### ISSUE THREAD

**Issue by ormsbee** [openedx/forum#137]
By default, we had intended for sites running master to require no changes at all to their configuration to keep the old experience. We currently break this, as @regisb notes in [this post](https://discuss.openedx.org/t/forum-v2-compatibility-on-open-edx-master-branch/14519). The code is currently doing this because we sometimes need to derive the course_id to know which code path to take, and that's not always available from the request itself.

Is it feasible to add checking to [`forum.backends.mongodb.api.get_course_id_by_thread_id`](https://github.com/openedx/forum/blob/36bdee1ca0525faffad81d528e7e88f6944c151b/forum/backends/mongodb/api.py#L1591) to explicitly check to see if `FORUM_MONGODB_DATABASE` and `FORUM_MONGODB_CLIENT_PARAMETERS` exist, and return `None` if they don't? That should make the public api call [`get_course_id_by_thread`](https://github.com/openedx/forum/blob/36bdee1ca0525faffad81d528e7e88f6944c151b/forum/api/threads.py#L392-L405) return None. (And the same for `get_course_id_by_comment`.)

Would that work? Are there other places where we need to try to infer the course_id?

**Comment by ormsbee** [comment]
FYI @taimoor-ahmed-1, @Ali-Salman29, @Faraz32123

**Comment by regisb** [comment]
> Is it feasible to add checking to [forum.backends.mongodb.api.get_course_id_by_thread_id](https://github.com/openedx/forum/blob/36bdee1ca0525faffad81d528e7e88f6944c151b/forum/backends/mongodb/api.py#L1591) to explicitly check to see if FORUM_MONGODB_DATABASE and FORUM_MONGODB_CLIENT_PARAMETERS exist, and return None if they don't?

There are two issues with this approach:

1. These settings have default values which are not `None`. We could remove those default values, but then that would mean more work for people enabling forum v2.
2. There are some cases when we need `get_course_id_by_thread_id` to not return `None`: that is, when forum v1 is enabled across the platform, but forum v2 is enabled only for select courses.

> Are there other places where we need to try to infer the course_id?

I believe these are the only two places, but @Faraz32123 might know better.

IMHO, the "right" fix would be to have the forum send the course ID along with the thread ID/comment ID in these HTTP requests. But this is no longer a trivial fix.

**Comment by Faraz32123** [comment]
> I believe these are the only two places, but @Faraz32123 might know better.

Yes, these are the 2 places in forumV2 where we can return None by checking settings. But This will result in above 2 issues. Even if we add this settings check in edx-platform, issue 2 will still remain when forum v2 is enabled only for selected courses.

> IMHO, the "right" fix would be to have the forum send the course ID along with the thread ID/comment ID in these HTTP requests. But this is no longer a trivial fix.

Exactly.

**Comment by ormsbee** [comment]
> 1. These settings have default values which are not None. We could remove those default values, but then that would mean more work for people enabling forum v2.

Maybe tutor could set those defaults? Or detect when `FORUM_MONGODB_CLIENT_PARAMETERS` hasn't been changed? (The default parameters shouldn't be deployed without a password, right?)

Really though, it doesn't have to be the settings values. I'm just looking for any signifier of "this connection isn't functional, so we'll never find a course, so let's return None".

> 2. There are some cases when we need get_course_id_by_thread_id to not return None: that is, when forum v1 is enabled across the platform, but forum v2 is enabled only for select courses.

I realize this is likely my own ignorance (or lack of sleep), but I'm not clear why this is a problem. I agree that we need it to return actual course_ids at times, but if the `forum` app has a non-functioning connection to MongoDB, then it can't be running in a mixed v1/v2 mode to MongoDB anyway. As long as we're changing only the MongoDB backend's `get_course_id_by_thread_id`, it should still preserve behavior if the MySQL v2 backend returns that it has the course.

**Comment by regisb** [comment]
> Maybe tutor could set those defaults? 

Yes, you're right, and this reminds me that we forgot to include those default parameters in tutor-forum.

> I'm just looking for any signifier of "this connection isn't functional, so we'll never find a course, so let's return None".

I understand. Yet, I'm reluctant to use this pattern, because it makes the codebase even more confusing than it already is. 

I think we can avoid some of that complexity by introducing a global `DISABLE_FORUM_V2` setting, which will be checked before course waffle flags. Wdyt?

> I realize this is likely my own ignorance (or lack of sleep), but I'm not clear why this is a problem. I agree that we need it to return actual course_ids at times, but if the forum app has a non-functioning connection to MongoDB, then it can't be running in a mixed v1/v2 mode to MongoDB anyway. As long as we're changing only the MongoDB backend's get_course_id_by_thread_id, it should still preserve behavior if the MySQL v2 backend returns that it has the course.

Yes, you're right again. This will _work_, but the fact that it's difficult for us to wrap our brains around that solution tells us that maybe we should avoid it... I think it's much clearer if the app crashes because of incorrect settings than trying to guess what's the initial user intent.

**Comment by ormsbee** [comment]
I’m okay with a DISABLE_FORUM_V2 setting.

**Comment by regisb** [comment]
OK let's do this then. How do you want to proceed? Should we open a PR in the edx-platform master branch? Do we need it today?

**Comment by ormsbee** [comment]
Please open a PR to the master branch that has the now-reverted changes that introduce v2 forums, and then add this commit on top of it. Please mention it in the [Slack thread](https://axim-collaborative.slack.com/archives/C05BH88RYG6/p1733761150037549?thread_ts=1733759644.993529&cid=C05BH88RYG6) where this is being discussed.

**Comment by regisb** [comment]
OK here we go: https://github.com/openedx/edx-platform/pull/36002
